website/content/docs/api-gateway.mdx

```diff
@@ -0,0 +1,281 @@
+
+When gateways are created from a `GatewayClass`, they use the parameters specified in the `GatewayClass` at the time of instantiation.
+If you want to propagate changes made to the `GatewayClass` after gateways have been created, then the implementation must document this behavior.
+The `gateway-exists-finalizer.gateway.networking.k8s.io` finalizer must also be added to a `GatewayClass` to ensure that a `GatewayClass` associated with a `Gateway` is not deleted while in use.
+
+Add the `kind: GatewayClass` option to the the gateway values file to declare a gateway class.
+The following example creates a gateway class called `test-gateway-class`.
+
+<CodeBlockConfig filename="gateway.yaml">
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: GatewayClass
+metadata:
+  name: test-gateway-class
+spec:
+  controllerName: 'hashicorp.com/consul-api-gateway-controller'
+  parametersRef:
+    group: api-gateway.consul.hashicorp.com
+    kind: GatewayClassConfig
+    name: test-gateway-class-config
+```
+
+</CodeBlockConfig>
+
+Refer to the Kubernetes Gateway API documentation for details about configuring gateway classes:
+<https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1alpha2.GatewayClass>
+
+### GatewayClassConfig
+
+The `GatewayClassConfig` object describes additional Consul API Gateway-related configuration parameters for the `GatewayClass`.
+
+Add the `kind: GatewayClassConfig` option to the gateway values file to declare a gateway class.
+The following example creates a gateway class called `test-gateway-class-config`.
+
+<CodeBlockConfig filename="gateway.yaml">
+
+```yaml
+apiVersion: api-gateway.consul.hashicorp.com/v1alpha1
+kind: GatewayClassConfig
+metadata:
+  name: test-gateway-class-config
+spec:
+  useHostPorts: true
+  logLevel: trace
+  consul:
+    scheme: https
+    caSecret: consul-ca-cert
+    ports:
+      http: 8501
+      grpc: 8502
+```
+
+</CodeBlockConfig>
+
+The following table describes the required parameters for the `spec` array:
+
+| Parameter      | Description                                                                                                                                          | Type    | Default |
+| -------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------- | ------- | ------- |
+| `useHostPorts` | Enables the gateway to use the host ports                                                                                                            | Boolean | `true`  |
+| `logLevel`     | Specifies the error reporting level for logs. You can specify the following values: `FATAL`, `ERROR`, `WARN`, `INFO`, `DEBUG`, `TRACE`, `ALL`, `OFF` | String  | `trace` |
+| `consul`       | Specifies ingress properties for Consul. See [Consul Configuration](#consul-configuration)                                                           | Array   | N/A     |
```

Comment:
Here's the list taken from https://github.com/hashicorp/consul-api-gateway/blob/main/config/crd/bases/api-gateway.consul.hashicorp.com_gatewayclassconfigs.yaml which we may want to link to

| Parameter      | Description                                                                                                                                          | Type    | Default |
| -------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------- | ------- | ------- |
| `consul.address` | The address of the consul server to communicate with in the gateway pod. If not specified, the pod will attempt to use a local agent on the host on which it is running. | String | N/A |
| `consul.authentication.account` | The Kubernetes service account to authenticate as. | String | N/A |
| `consul.authentication.managed` | Whether deployments should be run with "managed" service accounts created by the gateway controller. When this option is specified `consul.authentication.account` is ignored. | Boolean | `false` |
| `consul.authentication.method` | The Consul auth method used for initial authentication by consul-api-gateway. | String | N/A |
| `consul.authentication.namespace` | The Consul namespace to use for authentication. | String | N/A |
| `consul.ports.grpc` | The grpc port for Consul's xDS server. | Integer | `8502` |
| `consul.ports.http` | The port for Consul's HTTP server. | Integer | `8500` |
| `consul.scheme` | The scheme to use for connecting to Consul. Either "http" or "https" | String | `"http"` |
| `copyAnnotations.service` | List of annotations to copy to the gateway service. | Array | `["external-dns.alpha.kubernetes.io/hostname"]` |
| `image.consulAPIGateway` | The image to use for consul-api-gateway. | String | `"hashicorp/consul-api-gateway:RELEASE_VERSION"` |
| `image.envoy` | The image to use for Envoy. | String | `"envoyproxy/envoy:v1.19-latest"` |
| `logLevel` | Logging levels. One of "trace", "debug", "info", "warning", or "error". | String | `"info"` |
| `nodeSelector` | NodeSelector is a selector which must be true for the pod to fit on a node. Selector which must match a node's labels for the pod to be scheduled on that node. More info: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/ | Object | N/A |
| `serviceType` | Service Type string describes ingress methods for a service. One of "ClusterIP", "NodePort", or "LoadBalancer". | String | N/A |
| `useHostPorts` | If this is set, then the Envoy container ports are mapped to host ports. | Boolean | `false` |

If you want to separate the segments into their own configuration sections, just explode them by the `.`.